tlslite/utils/rsakey.py

```diff
@@ -87,6 +91,167 @@ def hashAndVerify(self, sigBytes, bytes):
         result2 = self.verify(sigBytes, prefixedHashBytes2)
         return (result1 or result2)
 
+    def MGF1(self, mgfSeed, maskLen, hAlg):
+        """Generate mask from passed-in seed.
+
+        This generates mask based on passed-in seed and output maskLen.
+
+        @type mgfSeed: L{bytearray}
+        @param mgfSeed: Seed from which mask will be generated.
+
+        @type maskLen: int
+        @param maskLen: Wished length of the mask, in octets
+
+        @rtype: L{bytearray}
+        @return: Mask
+        """
+        hashLen = getattr(hashlib, hAlg)().digest_size
+        if maskLen > (2 ** 32) * hashLen:
+            raise MaskTooLongError("Mask generating function failed")
+        T = bytearray()
+        end = (Poly1305.divceil(maskLen, hashLen))
+        for x in range(0, end):
+            C = numberToByteArray(x, 4)
+            T += secureHash(mgfSeed + C, hAlg)
+        return T[:maskLen]
+
+    def EMSA_PSS_encode(self, M, emBits, hAlg, sLen=0):
+        """Encode the passed in message
+
+        This encodes the message using selected hash algorithm
+
+        @type M: bytearray
+        @param M: Message to be encoded
+
+        @type emBits: int
+        @param emBits: maximal length of returned EM
+
+        @type hAlg: str
+        @param hAlg: hash algorithm to be used
+
+        @type sLen: int
+        @param sLen: length of salt"""
+        hashLen = getattr(hashlib, hAlg)().digest_size
+        mHash = secureHash(M, hAlg)
+        emLen = Poly1305.divceil(emBits, 8)
+        if emLen < hashLen + sLen + 2:
+            raise EncodingError("Encoding error")
+        salt = getRandomBytes(sLen)
+        M2 = bytearray(8) + mHash + salt
+        H = secureHash(M2, hAlg)
+        PS = bytearray(emLen - sLen - hashLen - 2)
+        DB = PS + bytearray(b'\x01') + salt
+        dbMask = self.MGF1(H, emLen - hashLen - 1, hAlg)
+        maskedDB = bytearray(i ^ j for i, j in zip(DB, dbMask))
+        mLen = emLen*8 - emBits
+        mask = (1 << 8 - mLen) - 1
+        maskedDB[0] &= mask
+        EM = maskedDB + H + bytearray(b'\xbc')
+        return EM
+
+    def RSASSA_PSS_sign(self, M, hAlg, sLen=0):
+        """"Sign the passed in message
+
+        This signs the message using selected hash algorithm
+
+        @type M: bytearray
+        @param M: Message to be signed
+
+        @type hAlg: str
+        @param hAlg: hash algorithm to be used
+
+        @type sLen: int
+        @param sLen: length of salt"""
+        EM = self.EMSA_PSS_encode(M, numBits(self.n) - 1, hAlg, sLen)
+        m = bytesToNumber(EM)
+        if m >= self.n:
+            raise MessageTooLongError("Message too long")
```

Comment:
how about "encode output too long"? the message for the user it he M, while code is checking EM - that will be confusing to users